tasks/install_linux_repo.yml

```diff
@@ -0,0 +1,103 @@
+---
+# File: install_linux_repo.yml - package installation tasks for Consul
+
+- name: Install OS packages
+  package:
+    name: "{{ item }}"
+    state: present
+  with_items: "{{ consul_os_packages }}"
+  tags: installation
+
+- name: Populate service facts
+  ansible.builtin.service_facts:
+
+- name: Gather the package facts
+  ansible.builtin.package_facts:
+    manager: auto
+
+- name: Clean up previous consul data
+  block:
+    - name: Stop service consul, if running
+      ansible.builtin.systemd:
+        name: consul
+        state: stopped
+      when: ansible_facts.services['consul.service'] is defined
+
+    - name: Remove consul systemd unit file from previous installation
+      ansible.builtin.file:
+        path: /usr/lib/systemd/system/consul.service
+        state: absent
+      notify: systemctl daemon-reload
+
+    - name: Remove the user 'consul'
+      ansible.builtin.user:
+        name: consul
+        state: absent
+        remove: yes
+
+  when:
+    - "ansible_distribution|lower == 'redhat' or ansible_distribution|lower == 'centos' or \
+        ansible_distribution|lower == 'fedora' or ansible_distribution|lower == 'amazon' or \
+        ansible_distribution|lower == 'debian' or ansible_distribution|lower == 'ubuntu'"
+    - "'consul' not in ansible_facts.packages"
+
+- name: Install repository
+  block:
+    - name: Add Redhat/CentOS/Fedora/Amazon Linux repository
+      ansible.builtin.command: "yum-config-manager --add-repo {{ consul_repo_url }}"
+      args:
+        creates: /etc/yum.repos.d/hashicorp.repo
+      when: "ansible_distribution|lower == 'redhat' or ansible_distribution|lower == 'centos' or \
+            ansible_distribution|lower == 'fedora' or ansible_distribution|lower == 'amazon'"
+
+
+    - name: Add an Apt signing key, uses whichever key is at the URL
+      ansible.builtin.apt_key:
+        url: https://apt.releases.hashicorp.com/gpg
+        state: present
+      when: ansible_distribution|lower == 'debian' or ansible_distribution|lower == 'ubuntu'
+
+    - name: Add Redhat/CentOS/Fedora/Amazon Linux repository
```

Comment:
Please change this to Debian/Ubuntu

Reply:
Updated name to "Add Debian/Ubuntu Linux repository"